pytest.ini

```diff
@@ -4,4 +4,17 @@
 addopts=--strict-markers --tb=native -p pytester --runpytest=subprocess --durations=20 -rfEX
 xfail_strict = True
 filterwarnings =
+    error
     ignore::hypothesis.errors.NonInteractiveExampleWarning
+    # https://github.com/pandas-dev/pandas/issues/41199
+    default:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning
+    default:distutils Version classes are deprecated\. Use packaging\.version instead:DeprecationWarning
+    # https://github.com/pandas-dev/pandas/issues/32056 (?)
+    default:numpy\.ufunc size changed, may indicate binary incompatibility\. Expected 216 from C header, got 232 from PyObject:RuntimeWarning
+    # https://github.com/lark-parser/lark/pull/1140
+    default:module 'sre_constants' is deprecated:DeprecationWarning
+    default:module 'sre_parse' is deprecated:DeprecationWarning
+    # https://github.com/pandas-dev/pandas/issues/34848
+    default:`np\.bool` is a deprecated alias for the builtin `bool`:DeprecationWarning
+    default:`np\.complex` is a deprecated alias for the builtin `complex`:DeprecationWarning
+    default:`np\.object` is a deprecated alias for the builtin `object`:DeprecationWarning
```

Comment:
(no action required; I'll follow up on this later)

I'm inclined to drop support for Pandas < 1.1; we've kept it to date because it wasn't any additional maintenance hassle, but that's evidently changed.